fixes #431 Package publisher workflow

lpsandaruwan opened the pull request:
This workflow will check for version changes comparing the version with published artifacts in NPM package registry and will publish only the version changed packages.

A NPM token(for publishing package) need to be added to the Github repository secrets(named as `NPM_TOKEN`)  to use this CI workflow.
To generate NPM token, on [https://www.npmjs.com/](https://www.npmjs.com/)
1. In the upper right corner of the page, click your profile picture, then click Access Tokens.
2. Screenshot of the account menu with the tokens link selected
3. Click Create New Token.
4. Select the type of access token: Automation
5. Click Generate Token.
6. Copy the token from the top of page.
As explained in [https://docs.npmjs.com/creating-and-viewing-access-tokens](https://docs.npmjs.com/creating-and-viewing-access-tokens)

Then add it to the Github, as explained in [https://docs.github.com/en/actions/reference/encrypted-secrets](https://docs.github.com/en/actions/reference/encrypted-secrets)

1. On GitHub, navigate to the main page of the repository.
2. Under your repository name, click  Settings.
3. Repository settings button
4. In the left sidebar, click Secrets.
5. Click New repository secret.
6. Type a name for your secret in the Name input box. int this case it is `NPM_TOKEN`
7. Enter the value for your secret.
8. Click Add secret.

endel commented:
Hi @lpsandaruwan, thanks for the PR, this approach is looking good! 

I was trying to run it locally, though, and I'm getting this error:

```
/Users/endel/Projects/colyseus/colyseus/.github/scripts/node_modules/npm/lib/view.js:61
    if (data["dist-tags"].hasOwnProperty(version)) {
                          ^

TypeError: Cannot read property 'hasOwnProperty' of undefined
    at /Users/endel/Projects/colyseus/colyseus/.github/scripts/node_modules/npm/lib/view.js:61:27
```

(Node v16.1.0, NPM 7.11.2) 

If you need more info about the error just let me know, we can try to inspect it together 

---

A few suggestions that I have:
- Use `NPM_AUTH_TOKEN` env var instead of `NPM_TOKEN` (we already have an `NPM_AUTH_TOKEN` set for publishing 🙂)
- Skip "private" packages during the version comparison loop (some packages are either internal and/or not ready and flagged as "private", [such as the example package](https://github.com/colyseus/colyseus/blob/bdf43169e509e0fd2305cd4a3b3669e7bf733c76/packages/example/package.json#L3))

lpsandaruwan commented:
Hi @endel,
Thank you very much! I will investigate the above,  also will look for applying  your suggestions and will get back to you. :+1

lpsandaruwan commented:
Hi @endel,

The above error you got due to an old package `npmview`, I removed that dependency from workflow. Also added functionality to skip private packages and changed the NPM token variable name. Please review and let me know if it requires more changes.
